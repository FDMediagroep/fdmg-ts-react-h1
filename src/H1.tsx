import * as React from 'react';

export interface Props {
    className?: string;
    children?: JSX.Element|string|number;
}

/**
 * H1 compononent renders a heading
 */
export default class H1 extends React.Component<Props, any> {
    public state: any;
    public props: any;

    constructor(props: Props) {
        super(props);
        this.props = props;
        console.info('H1 component', props);
    }

    render() {
        return (
            <h1 className={this.props.className}>{this.props.children}</h1>
        );
    }
}
