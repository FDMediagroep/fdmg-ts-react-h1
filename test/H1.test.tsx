import * as React from 'react';
import * as renderer from 'react-test-renderer';
import H1 from '../src/H1';

console.info = function() {};

test('H1 renders correctly', () => {
    let component = renderer.create(
        <H1>test h1</H1>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <H1/>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <H1><span>testing 123</span><div>testing456</div></H1>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('H1 renders correctly with CSS class', () => {
    const component = renderer.create(
        <H1 className={'title'}>test h1</H1>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});