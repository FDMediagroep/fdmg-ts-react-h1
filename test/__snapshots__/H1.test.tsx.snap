// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 renders correctly 1`] = `
<h1>
  test h1
</h1>
`;

exports[`H1 renders correctly 2`] = `<h1 />`;

exports[`H1 renders correctly 3`] = `
<h1>
  <span>
    testing 123
  </span>
  <div>
    testing456
  </div>
</h1>
`;

exports[`H1 renders correctly with CSS class 1`] = `
<h1
  className="title"
>
  test h1
</h1>
`;
